test.js

```diff
@@ -211,3 +211,52 @@ test('throws useful error message when non-error is thrown', async t => {
 		message: /Non-error/,
 	});
 });
+
+if (globalThis.AbortController !== undefined) {
+	test('aborts with an AbortSignal', async t => {
+		t.plan(2);
+
+		let index = 0;
+		const controller = new AbortController();
+
+		await t.throwsAsync(pRetry(async attemptNumber => {
+			await delay(40);
+			index++;
+			if (attemptNumber === 3) {
+				controller.abort();
+			}
+
+			throw fixtureError;
+		}, {
+			signal: controller.signal,
+		}), {
+			instanceOf: globalThis.DOMException === undefined ? Error : DOMException,
```

Comment:
Should I duplicate the test case instead? (i.e. make one with `Error` and one with `DOMException`)

Reply:
This is fine. It's just a temporary thing anyway.